cloud/amazon/ecs_taskdefinition.py

```diff
@@ -120,14 +122,35 @@ def __init__(self, module):
                 module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
             self.ecs = boto3_conn(module, conn_type='client', resource='ecs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
         except boto.exception.NoAuthHandlerFound, e:
-            module.fail_json(msg="Can't authorize connection - "+str(e))
+            self.module.fail_json(msg="Can't authorize connection - "+str(e))
 
     def describe_task(self, task_name):
         try:
             response = self.ecs.describe_task_definition(taskDefinition=task_name)
             return response['taskDefinition']
-        except botocore.exceptions.ClientError:
-            return None
+        except botocore.exceptions.ClientError, e:
+            self.module.fail_json(msg="Can't describe task - "+str(e))
+
+    def is_matching_task(self, expected, existing):
+        expectedContainers = expected['containers']
+
+        for container in expectedContainers:
+            # in case those properties are not defined, we put the default values
+            # to match the result from Amazon
+            for list_prop in ['environment', 'mountPoints', 'portMappings', 'volumesFrom']:
+                if list_prop not in container:
+                    container[list_prop] = []
+
+            for mount_point in container['mountPoints']:
+                if 'readOnly' not in mount_point:
+                    mount_point['readOnly'] = 'false'
```

Comment:
`readOnly` must be a boolean[1]. This causes the following error:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Invalid type for parameter containerDefinitions[0].mountPoints[0].readOnly, value: false, type: <type 'str'>, valid types: <type 'bool'>
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "parsed": false}
```

The fix is:

```python
mount_point['readOnly'] = False
```

[1] http://boto3.readthedocs.org/en/latest/reference/services/ecs.html#ECS.Client.register_task_definition